src/components/TimeRestrictionsTable.tsx

```diff
@@ -8,77 +8,138 @@ interface TimeRestrictionsTableProps {
-                                errorClass="govuk-input--error"
-                            >
-                                <>
-                                    <label
-                                        className={`govuk-label ${index === 0 ? '' : 'govuk-visually-hidden'}`}
-                                        htmlFor={`end-time-${chosenDay}`}
-                                    >
-                                        End time <span className="govuk-visually-hidden">for {chosenDay}</span>
-                                    </label>
+                    <div>
+                        <FormGroupWrapper errors={errors} errorId={`end-time-${chosenDay}-${i}`}>
+                            <>
+                                <label className="govuk-label" htmlFor={`end-time-${chosenDay}`}>
+                                    End time <span className="govuk-visually-hidden">for {chosenDay}</span>
+                                </label>
+                                <FormElementWrapper
+                                    errors={errors}
+                                    errorId={`end-time-${chosenDay}-${i}`}
+                                    errorClass="govuk-input--error"
+                                >
                                     <input
-                                        className="govuk-input govuk-input--width-4"
-                                        id={`end-time-${chosenDay}`}
+                                        className="govuk-input govuk-input--width-5"
+                                        id={`end-time-${chosenDay}-${i}`}
                                         name={`endTime${chosenDay}`}
                                         aria-describedby="time-restrictions-hint"
                                         type="text"
-                                        defaultValue={findCorrectDefaultValue(endTimeInputs, chosenDay)}
+                                        defaultValue={findCorrectDefaultValue(endInputsForDay, chosenDay, i)}
                                     />
-                                </>
-                            </FormElementWrapper>
+                                </FormElementWrapper>
+                            </>
                         </FormGroupWrapper>
                     </div>
-                </fieldset>
+                </div>,
+            );
+        }
+
+        return rows;
+    };
+
+    return (
+        <>
+            {chosenDays.map(chosenDay => (
+                <>
+                    <fieldset className="govuk-fieldset flex-container time-restrictions-table" key={chosenDay}>
+                        <legend className="govuk-fieldset__legend govuk-fieldset__legend--m day-label">
+                            {sentenceCaseString(chosenDay)}
+                        </legend>
+                        <div>{getTimeRestrictionRows(chosenDay)}</div>
+                        <button
```

Comment:
It might be good to only render the button when there are fewer than 5 time bands to prevent any confusion:

```
{dayRowCount[chosenDay] < 5 && (
      <button
          id={`add-another-button-${chosenDay}`}
          type="button"
          className="govuk-button govuk-button--secondary govuk-!-margin-left-3 time-restrictions-button-placement"
          onClick={(): void => addTimeRestrictionRow(chosenDay)}
      >
          Add another timeband
      </button>
  )}
```